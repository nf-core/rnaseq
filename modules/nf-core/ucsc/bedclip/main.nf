process UCSC_BEDCLIP {
    tag "$meta.id"
    label 'process_medium'

    // WARN: Version information not provided by tool on CLI. Please update version string below when bumping container versions.
    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/ucsc-bedclip:377--h0b8a92a_2' :
        'biocontainers/ucsc-bedclip:377--h0b8a92a_2' }"

    input:
    tuple val(meta), path(bedgraph)
    path  sizes

    output:
    tuple val(meta), path("*.bedGraph"), emit: bedgraph
    path "versions.yml"                , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '377' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    bedClip \\
        $bedgraph \\
        $sizes \\
        ${prefix}.bedGraph

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        ucsc: $VERSION
    END_VERSIONS
    """
}
