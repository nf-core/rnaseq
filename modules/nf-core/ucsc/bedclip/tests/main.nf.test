nextflow_process {

    name "Test Process UCSC_BEDCLIP"
    script "../main.nf"
    process "UCSC_BEDCLIP"
    tag "modules"
    tag "modules_nfcore"
    tag "ucsc"
    tag "ucsc/bedclip"

    test("sarscov2") {
        config "./nextflow.config"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + "genomics/sarscov2/illumina/bedgraph/test.bedgraph", checkIfExists: true)
                ])
                input[1] = Channel.of(file(params.modules_testdata_base_path + "genomics/sarscov2/genome/genome.sizes", checkIfExists: true))
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}