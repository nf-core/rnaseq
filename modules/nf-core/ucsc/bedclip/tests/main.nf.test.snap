{
    "sarscov2": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.clip.bedGraph:md5,e02395e1f7c593b3f79563067159ebc2"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b92248ce2a138ad4d146ac0d9e71fb34"
                ],
                "bedgraph": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.clip.bedGraph:md5,e02395e1f7c593b3f79563067159ebc2"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b92248ce2a138ad4d146ac0d9e71fb34"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-07T16:04:04.165478"
    }
}